### DISCUSSION TITLE
Git navigation capability toggle

### DISCUSSION THREAD

**Discussion by 20240816** [sveltia/sveltia-cms#243] [Ideas]
I'd like the ability to toggle whether the CMS user is shown options to navigate to git.
Maybe a user type setting with this being a 'developer' feature?

**Comment by kyoshino** [comment]
I guess you mean these menu items?

<img width="205" alt="image" src="https://github.com/user-attachments/assets/b90b81b2-f67b-4019-844e-667141edc77e">

<img width="204" alt="image" src="https://github.com/user-attachments/assets/52b4566c-12e6-4905-af2d-11e8a0bfd952">

<img width="201" alt="image" src="https://github.com/user-attachments/assets/ea469839-11a5-48d8-bfbe-b8e5d58f6d80">

I was thinking about making the change, because end-users are not necessarily developers. (Currently users have to sign in with GitHub/GitLab anyway but that’s another story.)

There is already Developer Mode that can be enabled via the app settings, so I’ll show these options according to it.

**Reply by 20240816** [reply]
Yes exactly :smile:

**Reply by 20240816** [reply]
This approach will still allow the git user full repo access (e.g. via GitHub directly) so to truly restrict wider repo access I've gone with the approach of relating the CMS directly to a new submodule repo which I've added to the main repo `git submodule add [url here] [submodule name here]` and kept updated on compile time in Cloudflare via prepending the Cloudflare build command with `git submodule update --remote --merge;` The .gitmodules file also needed the url in the format `url = git@github.com:{user}/{submodule repo}.git`

Having this toggle option will still be useful but CMS developers should understand that full repo access remains possible.

**Comment by kyoshino** [comment]
Fixed in [v0.45.3](https://github.com/sveltia/sveltia-cms/releases/tag/v0.45.3) ✌🏼
